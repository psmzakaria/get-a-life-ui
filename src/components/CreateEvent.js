import React, { Component } from "react";
import { Button, Form, Modal, Card, Icon } from "semantic-ui-react";

class CreateEvent extends Component {
  render() {
    return (
      <div>
        <Modal
          trigger={
            <div className="esnap-container-size">
              <Card onClick={this.props.handleOpen}>
                <div className="esnap-icon">
                  <Icon name="add to calendar" size="huge" />
                </div>
                <Card.Content>
                  <Card.Header>Create New Event</Card.Header>
                </Card.Content>
              </Card>
            </div>
          }
          open={this.props.modalOpen}
          onClose={this.props.handleClose}
          closeIcon
        >
          <Modal.Header>Create Your Event</Modal.Header>
          <Modal.Content image scrolling>
            <Modal.Description>
              <Form>
                <Form.Field>
                  <label>Title</label>
                  <input
                    id="title"
                    placeholder="Title of the event"
                    value={this.props.formFields.title}
                    onChange={event => this.props.handleChange(event, "title")}
                  />
                </Form.Field>
                <Form.Field>
                  <label>Start Date</label>
                  <input
                    id="start"
                    placeholder="DD/MM/YYYY"
                    value={this.props.formFields.startDate}
                    onChange={event =>
                      this.props.handleChange(event, "startDate")
                    }
                  />
                </Form.Field>
                <Form.Field>
                  <label>End Date</label>
                  <input
                    id="end"
                    placeholder="DD/MM/YYYY"
                    value={this.props.formFields.endDate}
                    onChange={event =>
                      this.props.handleChange(event, "endDate")
                    }
                  />
                </Form.Field>
                <Button type="submit" onClick={this.props.handleSubmit}>
                  Submit
                </Button>
              </Form>
            </Modal.Description>
          </Modal.Content>
        </Modal>
      </div>
    );
  }
}

export default CreateEvent;
