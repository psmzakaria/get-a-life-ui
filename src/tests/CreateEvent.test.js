import React from "react";
import CreateEvent from "./../components/CreateEvent";

import ShallowRenderer from "react-test-renderer/shallow";

it("renders the book component of the app ", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<CreateEvent />);
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
