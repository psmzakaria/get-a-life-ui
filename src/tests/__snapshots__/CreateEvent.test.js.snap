// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the create component of the app  1`] = `
<div>
  <Modal
    centered={true}
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={false}
    trigger={
      <div
        className="esnap-container-size"
      >
        <Card
          onClick={[Function]}
        >
          <div
            className="esnap-icon"
          >
            <Icon
              as="i"
              name="add to calendar"
              size="huge"
            />
          </div>
          <CardContent>
            <CardHeader>
              Create New Event
            </CardHeader>
          </CardContent>
        </Card>
      </div>
    }
  >
    <ModalHeader>
      Create Your Event
    </ModalHeader>
    <ModalContent
      image={true}
      scrolling={true}
    >
      <ModalDescription>
        <Form
          as="form"
        >
          <FormField>
            <label>
              Title
            </label>
            <input
              id="title"
              onChange={[Function]}
              placeholder="Title of the event"
              value=""
            />
          </FormField>
          <FormField>
            <label>
              Start Date
            </label>
            <input
              id="start"
              onChange={[Function]}
              placeholder="DD/MM/YYYY"
              value=""
            />
          </FormField>
          <FormField>
            <label>
              End Date
            </label>
            <input
              id="end"
              onChange={[Function]}
              placeholder="DD/MM/YYYY"
              value=""
            />
          </FormField>
        </Form>
      </ModalDescription>
    </ModalContent>
    <ModalActions>
      <CreateInvitationModal
        handleSubmit={[Function]}
      />
    </ModalActions>
  </Modal>
</div>
`;
